sound/soc/sof/intel/cnl.c

```diff
@@ -100,10 +96,17 @@ static irqreturn_t cnl_ipc_irq_thread(int irq, void *context)
 
 		cnl_ipc_host_done(sdev);
 
-		ret = IRQ_HANDLED;
+		ipc_irq = true;
 	}
 
-	return ret;
+	if (!ipc_irq)
+		return IRQ_NONE;
```

Comment:
Go without re-enable IPC interrupt? This looks NOK for me.

Reply:
@keyonjie rethinking about this, it seems illogical to me to re-enable an interrupt and return IRQ_NONE. IRQ_NONE/IRQ_HANDLED are used to deal with shared interrupts and make sure that you either let the next handler deal with the interrupt (IRQ_NONE) or signal that the current handler took care of things (IRQ_HANDLED). If you re-enable an interrupt AND return IRQ_NONE, you would not respect the design of the IRQ framework.

I also wonder if this IRQ_NONE case can happen at all since we take care of the sharing at the initial interrupt level by checking it's an IPC interrupt.

Reply:
Agree, I don't think this IRQ_NONE(IPC interrupts that we don't handle here) case can happen, even it happen, since nobody else in current code will handle it, so it's good enough to return HANDLED and re-enable IPC interrupt.

Reply:
when irq_thread function is invoked, the  ADSPIC is disabled. So it is better to enable it at the end of the this function, at least generating a error message to notify us for error case.

Reply:
@keyonje @RanderWang are you saying IRQ_NONE should not be a return status since it can't happen, and that we should re-enable the IRQ unconditionally, with an error message logged if none of the new message or reply cases were handled. If yes, that's a reasonable solution.

Reply:
yes, I agree with your idea